blivet/devicetree.py

```diff
@@ -883,9 +884,25 @@ def unhide(self, device):
                 lvm.lvm_cc_removeFilterRejectRegexp(hidden.name)
 
     def _is_ignored_disk(self, disk):
-        return ((self.ignored_disks and disk.name in self.ignored_disks) or
-                (self.exclusive_disks and
-                 disk.name not in self.exclusive_disks))
+        """ Checks config for lists of exclusive and ignored disks
+            and returns if the given one should be ignored
+        """
+
+        def disk_in_taglist(disk, taglist):
+            # Taglist is a list containing mix of disk names and tags into which disk may belong.
+            # Check if it does. Raise ValueError if unknown tag is encountered.
+            if disk.name in taglist:
+                return True
+            tags = [t[1:] for t in taglist if t.startswith("@")]
+            for tag in tags:
+                if tag not in Tags.__members__:
+                    raise ValueError("unknown ignoredisk tag '@%s' encountered" % tag)
+                if Tags(tag) in disk.tags:
+                    return True
+            return False
+
+        return ((self.ignored_disks and disk_in_taglist(disk, self.ignored_disks)) or
+                (self.exclusive_disks and not disk_in_taglist(disk, self.exclusive_disks)))
```

Comment:
Separating the logic for matching devices by name from matching them by tags would make it easier to see what's going on in this code. You could also probably streamline it into something like this (starting after line 896):

```python
if not all(tag in Tags.__members__ for tag in tags):
    raise ValueError("invalid tag specified")
return any(Tags(tag) in disk.tags for tag in tags)
```